core/controlplane/config/templates/cluster.yaml

```diff
@@ -1071,6 +1071,43 @@ addons:
 #   # When enabled, `kubectl drain` is run on the node going to shut-down
 #   nodeDrainer:
 #     enabled: true
+#   # Enable dex  integration - https://github.com/coreos/dex
+#   # Configure OpenID Connect token authenticator plugin in Kubernetes API server. Notice: always use "https" for the "url",
+#   # otherwise the Kubernetes API server will not start correctly. The "caFile" can be skipped if you use a certificate signed
+#   # by a trusted CA.
+#   dex:
+#     enabled: true
+#     url: "https://dex.example.com"
```

Comment:
As I already added the functionality to generate the certificates from `kube-aws` I think maybe It will be  better to configure `dex`with https on port 5556 and create a NodePort service. Then add a ELB and a record set.